llvm/test/CodeGen/AIE/GlobalISel/xfail-legalize-shufflevector.mir

```diff
@@ -28,48 +28,3 @@ body:             |
     %0:_(<64 x s1>) = G_SHUFFLE_VECTOR %1(<64 x s1>), %1, shufflemask(undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef)
     G_STORE %0(<64 x s1>), %2(p0) :: (store (<64 x s1>) into `<64 x i1>* undef`, align 2)
     PseudoRET implicit $lr
-...
-
----
-name:            f_32x8
-body:             |
-  bb.0:
-    %1:_(<32 x s8>) = G_IMPLICIT_DEF
-    %2:_(p0) = G_IMPLICIT_DEF
-    %0:_(<32 x s8>) = G_SHUFFLE_VECTOR %1(<32 x s8>), %1, shufflemask(undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef)
-    G_STORE %0(<32 x s8>), %2(p0) :: (store (<32 x s8>) into `<32 x i8>* undef`, align 2)
-    PseudoRET implicit $lr
-...
-
----
-name:            f_16x16
-body:             |
-  bb.0:
-    %1:_(<16 x s16>) = G_IMPLICIT_DEF
-    %2:_(p0) = G_IMPLICIT_DEF
-    %0:_(<16 x s16>) = G_SHUFFLE_VECTOR %1(<16 x s16>), %1, shufflemask(undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef)
-    G_STORE %0(<16 x s16>), %2(p0) :: (store (<16 x s16>) into `<16 x i16>* undef`, align 2)
-    PseudoRET implicit $lr
-...
-
----
-name:            f_32x16
-body:             |
-  bb.0:
-    %1:_(<32 x s16>) = G_IMPLICIT_DEF
-    %2:_(p0) = G_IMPLICIT_DEF
-    %0:_(<32 x s16>) = G_SHUFFLE_VECTOR %1(<32 x s16>), %1, shufflemask(undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef, undef)
-    G_STORE %0(<32 x s16>), %2(p0) :: (store (<32 x s16>) into `<32 x i16>* undef`, align 2)
-    PseudoRET implicit $lr
-...
-
----
-name:            f_2x64
```

Comment:
Super nit: this currently fails in
```
bool llvm::AIELegalizerInfo::legalizeG_BUILD_VECTOR(llvm::LegalizerHelper &, llvm::MachineInstr &) const: Assertion `(EltSize == 8 || EltSize == 16 || EltSize == 32) && "non-existent integer size"' failed.
```

Reply:
This doesn't fail, it just seems like it would. The precombiner does some magic and turns it into a copy, which can be legalized completely.

Snippet from the output using commit e503a9c8a65c2a0236d891f64c06b2b5795971ac
```
Erasing: %0:_(<2 x s64>) = G_SHUFFLE_VECTOR %1:_(<2 x s64>), %1:_, shufflemask(undef, undef)

5203: GIR_Done
Created: %0:_(<2 x s64>) = COPY %1:_(<2 x s64>)                                                                                                            
Try combining %0:_(<2 x s64>) = COPY %1:_(<2 x s64>)
10: GIM_SwitchOpcode(MIs[0], [19, 225), Default=5243, JumpTable...) // Got=19
835: Begin try-block
842: GIM_CheckSimplePredicate(Predicate=3)
846: GIM_CheckCxxPredicate(MIs[0], Predicate=3)
849: GIR_CustomAction(FnID=3)
Erasing: %0:_(<2 x s64>) = COPY %1:_(<2 x s64>)
```

Reply:
The pre-legalizer combiner has nothing to do with this, this test runs the legalizer.
Please add it back, then this PR can be merged

Reply:
Sure will do, it is strange because I did run the update script multiple times before committing the change. I'll put that down to "works on my machine"

Reply:
It is added back to the commit